Packs/FireEyeETP/Integrations/FireEyeETPEventCollector/FireEyeETPEventCollector.py

```diff
@@ -641,17 +721,86 @@ def _get_max_events_to_fetch(params_max_fetch: str | int, arg_limit: str | int)
+
+    Returns: 'oauth2' for Client ID/Secret, 'api_key' for API Key
+    Raises: ValueError if authentication configuration is invalid or over-configured.
+    """
+    # Use function arguments instead of global variables
+    has_client_id = bool(client_id)
+    has_client_secret = bool(client_secret)
+    has_api_key = bool(api_key)
+    has_scopes = bool(scope)
+
+    # 1. CHECK FOR AMBIGUOUS OVER-CONFIGURATION
+    if has_client_id and has_client_secret and has_api_key:
+        raise ValueError(
+            "Both OAuth2 (Client ID/Secret) and API Key were provided. " "Please configure only one authentication method."
+        )
+
+    # 2. OAUTH2 VALIDATION
+    if has_client_id and has_client_secret:
+        # Check for required SCOPES when using OAuth2
+        if not has_scopes:
+            raise ValueError(
+                "Client ID and Client Secret provided, but the 'OAuth Scopes' parameter is missing. "
+                "Scopes are required for OAuth2 authentication."
+            )
+
+        # Assuming LOG_LINE is defined in your script environment
+        demisto.debug(f"{LOG_LINE} Authentication: Using OAuth2 (Client ID/Secret)")
+        return "oauth2"
+
+    # 3. API KEY VALIDATION
+    if has_api_key and not has_client_id and not has_client_secret:
+        demisto.debug(f"{LOG_LINE} Authentication: Using API Key")
+        return "api_key"
+
+    # 4. INCOMPLETE OAUTH2 CONFIGURATION
+    if has_client_id and not has_client_secret:
+        raise ValueError(
+            "Client ID provided but Client Secret is missing. "
+            "Both Client ID and Client Secret are required for OAuth2 authentication."
+        )
+
+    if has_client_secret and not has_client_id:
+        raise ValueError(
+            "Client Secret provided but Client ID is missing. "
+            "Both Client ID and Client Secret are required for OAuth2 authentication."
+        )
+
+    # 5. NO AUTHENTICATION METHOD PROVIDED
+    raise ValueError("No authentication credentials provided.")
+
+
 def main() -> None:  # pragma: no cover
     params = demisto.params()
     args = demisto.args()
 
+    # Extract authentication parameters - prioritize OAuth2 over legacy API Key
+    client_id = params.get("oauth_credentials", {}).get("identifier", "")
+    client_secret = params.get("oauth_credentials", {}).get("password", "")
+    scope = params.get("oauth_scopes", "etp.conf.ro etp.rprt.ro").strip()
     api_key = params.get("credentials", {}).get("password", "")
     base_url = params.get("url", "").rstrip("/")
     verify = not params.get("insecure", False)
     proxy = params.get("proxy", False)
     outbound_traffic = argToBoolean(params.get("outbound_traffic", False))
     hide_sensitive = argToBoolean(params.get("hide_sensitive", True))
 
+    # Validate authentication configuration
+    try:
+        validate_authentication_params(client_id, client_secret, api_key, scope)
+    except ValueError as e:
+        return_error(str(e))
```

Comment:
Why doing it seperatly than other calls? we rather not have systemexit in the middle of the code.